firebase-storage/ktx/src/main/kotlin/com/google/firebase/storage/ktx/StorageProgress.kt

```diff
@@ -0,0 +1,16 @@
+package com.google.firebase.storage.ktx
+
+/**
+ * Used to emit events about the progress of storage tasks.
+ */
+abstract class StorageProgress<T> private constructor() {
```

Comment:
Not using a `sealed class` here to avoid binary incompatibility in a future release if some day we potentially include a new listener/event as part of progress monitoring.

Binary incompatibility is well explained here: https://jakewharton.com/public-api-challenges-in-kotlin/#binary-compatibility